diesel/src/macros/mod.rs

```diff
@@ -45,6 +51,26 @@ macro_rules! column {
         {
         }
 
+        impl<Left, Right, FK> $crate::expression::SelectableExpression<
```

Comment:
I double checked that I wasn't missing something, and I'm not. I've explored this change *numerous* times. We cannot write

```
impl<Other> SelectableExpression<InnerJoinSource<table, Other>> for column where
    Other: Table,
    table: JoinTo<Other, Inner>,
{}

impl<Other> SelectableExpression<InnerJoinSource<Other, table>> for column where
    Other: Table + JoinTo<table, Inner>,
{}
```

because they overlap if `Other == table`. For your impl to be correct, you'd need to add a `SelectableExpression` constraint, which would be:

```rust
impl<Left, Right> SelectableExpression<InnerJoinSource<Left, Right>> for column where
    Left: JoinTo<Right, Inner>,
    Right: Table,
    column: SelectableExpression<Left>,
{}

impl<Left, Right> SelectableExpression<InnerJoinSource<Left, Right>> for column where
    Left: JoinTo<Right, Inner>,
    Right: Table,
    column: SelectableExpression<Right>,
{}
```

which again, overlaps if `Left` and `Right` are the same type, or if `column: SelectableExpression<Left> + SelectableExpression<Right>`. There's no way to properly resolve this in the language today without a more in depth rethink of this code.